### DISCUSSION TITLE
rnmapbox + ignite

### DISCUSSION THREAD

**Discussion by thertzelle** [infinitered/ignite#2311] [Q&A]
I've been struggling all afternoon to understand how to use rnmapbox + ignite (https://github.com/rnmapbox/maps).
I follow the instructions they have [here](https://github.com/rnmapbox/maps/blob/main/plugin/install.md) by installing the package with expo, then adding the information to plugin. But then their message "Next, rebuild your app as described in the ["Adding custom native code"](https://docs.expo.io/workflow/customizing/) guide." is really vague. 

When I tried to run `npx expo run:ios` on the ignite project, I immediately get a message:

```
Error: [ios.expoPlist]: withIosExpoPlistBaseMod: ENOENT: no such file or directory, open '/Users/username/Development/Project/ios/Project/Supporting/Expo.plist'
```
What am I doing wrong? How can I install extra package dependencies and use ignite ?

**Comment by frankcalise** [comment]
@thertzelle Hey thanks for your interest in Ignite!

What you'll have to do to utilize that package is build your own custom dev client (either locally or through EAS). This is the process that will enable that plugin for the native code you are looking to use and stay with Expo

Then once you have that dev client, you can install it on your simulator or device and run with `npx expo start --dev-client`

We have some guides on how to do this over at the Ignite Cookbook! Check out the guide here: https://infinitered.github.io/ignite-cookbook/docs/StayingWithExpo

Hope that helps, feel free to ask any follow up questions or reach out to me on the community Slack!

**Reply by thertzelle** [reply]
Thanks for the information. I've not been able to get the ignite project to `expo eject` or `npx expo prebuild` or run with eas, as I keep getting the same error message:

```
Error: [ios.expoPlist]: withIosExpoPlistBaseMod: ENOENT: no such file or directory, open '/Users/username/Development/Project/ios/Project/Supporting/Expo.plist'
```

**Comment by frankcalise** [comment]
@thertzelle that looks to be an issue with the config plugin they developed possibly.

You can try making sure your project is Expo only with these steps and see if it clears anything up though:

https://infinitered.github.io/ignite-cookbook/docs/PristineExpoProject

`expo eject` is dead so no need to use that. 

`npx expo prebuild` will generate the native directories for you with the config changes generated. you can then use `npx expo run:ios` for example to test it out. Keep in mind that's kind of the equivalent of ejecting in today's Expo, but you can certainly use it as a test to make sure your library is working with the app - then if it is, continue to build the custom dev client.
